lib/Utils.vala

```diff
@@ -325,8 +312,8 @@ namespace Gala {
          *
          * @return the close button pixbuf or null if it failed to load
          */
-        public static Gdk.Pixbuf? get_close_button_pixbuf () {
-            var height = 36 * Utils.get_ui_scaling_factor ();
+        public static Gdk.Pixbuf? get_close_button_pixbuf (float scale_factor) {
+            var height = (int)(36 * scale_factor);
             if (close_pixbuf == null || close_pixbuf.height != height) {
                 try {
                     close_pixbuf = new Gdk.Pixbuf.from_resource_at_scale (
```

Comment:
As we might have mixed DPI support in some time, it might make sense to use a HashTable of <height, Pixbuf>?